src/goto-programs/goto_program.h

```diff
@@ -529,6 +529,23 @@ class goto_programt
     return t;
   }
 
+  optionalt<const_targett> get_target(const unsigned location_number) const
+  {
+    PRECONDITION(!instructions.empty());
+
+    const unsigned start_location_number = instructions.front().location_number;
+
+    if(
+      location_number < start_location_number ||
+      location_number > instructions.back().location_number)
+    {
+      return nullopt;
+    }
+
+    return std::next(
+      instructions.begin(), location_number - start_location_number);
```

Comment:
I'm not sure we provide a guarantee that location numbers are contiguous. You might want to add a `CHECK_RETURN` to make sure what you are about to return has the expected `location_number`.

Reply:
Check added.